### PR TITLE
feat (#114): implement animation to indicate dragged nodes interactivity

### PR DESCRIPTION
**Pull request by philippe-gagne**
Key changes: 
- Nodes now darken by 5% when hovered.
- When a node is being dragged and held over another node, the lower node has its border highlighted to indicate interactivity.
- aabb.js renamed to collisionDetection.js

### CODE REVIEW

File: mikado-app/src/viewmodel/displayLayerStore.js [hunk 1]
```diff
@@ -419,6 +424,22 @@ class DisplayLayerOperations {
     });
   }
 
+  highlightOrUnhighlightNode(target) {
+    if (target == null) { // if no target, unhighlight all nodes
```
Comment (danielzgtg) [inline mikado-app/src/viewmodel/displayLayerStore.js:428]: `===` is preferred over `==`

File: mikado-app/src/viewmodel/displayLayerStore.js [hunk 2]
```diff
@@ -419,6 +424,22 @@ class DisplayLayerOperations {
...
+      for (const node of this.#state.nodes) { 
+        document.querySelector(`[data-id="${node.id}"]`).style.border = "1px solid rgba(105, 105, 105, 0.7)";
+      }
+    } else {
+      for (const node of this.#state.nodes) { // highlight target and unhighlight all other nodes
+        if (node.id != target.id){
```
Comment (danielzgtg) [inline mikado-app/src/viewmodel/displayLayerStore.js:434]: `!==` is preferred over `!=`

File: mikado-app/src/viewmodel/displayLayerStore.js [hunk 3]
```diff
@@ -125,6 +125,11 @@ class DisplayLayerOperations {
    */
   #backwardConnections = {};
 
+  /**
+   * ID of the node that is currently impending connection, or that was most recently impending connection.
+   */
+  #impendingConnectionNodeID = null;
```
Comment (danielzgtg) [inline mikado-app/src/viewmodel/displayLayerStore.js:131]: Where is this variable being used?

Comment (philippe-gagne) [reply mikado-app/src/viewmodel/displayLayerStore.js:131]: Oops, ended up not needing it but forgot to remove.

**Review by danielzgtg — APPROVED** [review]
LGTM

**Pull request merged**